atlasdb-tests-shared/src/test/java/com/palantir/atlasdb/cleaner/AsyncPuncherTest.java

```diff
@@ -82,34 +74,48 @@ public void delegatesInitializationCheck() {
     @Test
     public void testPuncherDurability() throws Exception {
         long stored = timestampService.getFreshTimestamp();
-        asyncPuncher.punch(stored);
-        checkExpectedValue(asyncPuncher, stored);
+        AsyncPuncher asyncPuncher = setupPuncher(THROWING_BACKUP_TIMESTAMP_SUPPLIER);
+        try {
+            asyncPuncher.punch(stored);
+            checkExpectedValue(asyncPuncher, stored);
+        } finally {
+            asyncPuncher.shutdown();
+        }
     }
 
 
     @Test
     public void testPuncherTimestampLessThanFreshTimestamp() throws Exception {
         long stored = timestampService.getFreshTimestamp();
-        asyncPuncher.punch(stored);
-        long retrieved = Long.MIN_VALUE;
-        for (int i = 0; i < MAX_INTERVALS_TO_WAIT && retrieved < stored; i++) {
-            Thread.sleep(ASYNC_PUNCHER_INTERVAL);
-            retrieved = asyncPuncher.getTimestampSupplier().get();
+        AsyncPuncher asyncPuncher = setupPuncher(THROWING_BACKUP_TIMESTAMP_SUPPLIER);
+        try {
+            asyncPuncher.punch(stored);
+            long retrieved = Long.MIN_VALUE;
+            for (int i = 0; i < MAX_INTERVALS_TO_WAIT && retrieved < stored; i++) {
+                Thread.sleep(ASYNC_PUNCHER_INTERVAL);
+                retrieved = asyncPuncher.getTimestampSupplier().get();
+            }
+            long freshTimestamp = timestampService.getFreshTimestamp();
+            assertThat(retrieved).isLessThan(freshTimestamp);
+        } finally {
+            asyncPuncher.shutdown();
         }
-        long freshTimestamp = timestampService.getFreshTimestamp();
-        assertThat(retrieved).isLessThan(freshTimestamp);
     }
 
     @Test
     public void punchesBackupTimestampWhenNothingWasPunched() throws Exception {
-        setupPuncher(timestampService::getFreshTimestamp);
```

Comment:
:hot_pepper: :hot_pepper: when it was called here

Reply:
I suppose this will spam `bla`: `No timestamp was found and attempting to get a fresh timestamp to punch failed.`